tasks/main.yml

```diff
@@ -65,3 +65,25 @@
       description:  "{{ item.value.description | default('<none>') }}"
       name:  "{{ item.value.name | default('<none>') }}"
       config: "{{ item.value.config | default([]) }}"
+
+- name: Write running configuration to startup configuration
+  local_action:
+    module: expect
+    command: "ssh {{ ansible_ssh_common_args | default }} {{ dell_powerconnect_switch_provider.username }}@{{ dell_powerconnect_switch_provider.host }}"
+    responses: >-
+      {{ dell_powerconnect_auth_responses |
+         combine(dell_powerconnect_write_responses) |
+         combine(enable_responses) |
+         combine(main_responses) }}
+  register: result
+  failed_when: >-
+    result.get('rc') != 255 or
+    '% Unrecognized command' in result.stdout_lines or
+    "% Invalid input detected at '^' marker." in result.stdout_lines or
+    "% missing mandatory parameter" in result.stdout_lines or
+    "% bad parameter value" in result.stdout_lines or
+    'Connection to ' ~ dell_powerconnect_switch_provider.host ~ ' closed.' not in result.stdout_lines[-1]
+  vars:
+    enable_responses: "{{ {dell_powerconnect_enable_prompt: ['enable', 'exit']} }}"
+    main_responses: "{{ {dell_powerconnect_main_prompt: [write_command, 'exit']} }}"
+  when: write_memory
```

Comment:
If you pass this through a `bool` filter it will accept yes/no etc.

Reply:
Thank you for letting me know about this! I had assumed the yes/no behavior was implicit; it's good to find out that my assumption was wrong. The bool filter has been added in the latest commit.